autokeras/auto/processor.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+
+
+class OneHotEncoder:
+    """A class that can format data.
+
+    This class provides ways to transform data's classification label into
+    vector.
+
+    Attributes:
+          data: The input data
+          n_classes: The number of classes in the classification problem.
+          labels: The number of labels.
+          label_to_vec: Mapping from label to vector.
+          int_to_label: Mapping from int to label.
+    """
+
+    def __init__(self):
+        """Initialize a OneHotEncoder"""
+        self.data = None
+        self.n_classes = 0
+        self.labels = None
+        self.label_to_vec = {}
+        self.int_to_label = {}
+
+    def fit(self, data):
+        """Create mapping from label to vector, and vector to label."""
+        data = np.array(data).flatten()
+        self.labels = set(data)
+        self.n_classes = len(self.labels)
+        for index, label in enumerate(self.labels):
+            vec = np.array([0] * self.n_classes)
+            vec[index] = 1
+            self.label_to_vec[label] = vec
+            self.int_to_label[index] = label
+
+    def transform(self, data):
+        """Get vector for every element in the data array."""
+        data = np.array(data)
+        if len(data.shape) > 1:
+            data = data.flatten()
+        return np.array(list(map(lambda x: self.label_to_vec[x], data)))
+
+    def inverse_transform(self, data):
+        """Get label for every element in data."""
+        return np.array(list(map(lambda x: self.int_to_label[x],
+                                 np.argmax(np.array(data), axis=1))))
+
+
+class Normalizer(object):
+    """ Perform basic image transformation and augmentation.
+
+    Attributes:
+        max_val: the maximum value of all data.
+        mean: the mean value.
+        std: the standard deviation.
+    """
+
+    def __init__(self):
+        self.max_val = None
+        self.mean = None
+        self.std = None
+
+    def fit(self, data):
+        self.max_val = data.max()
+        data = data / self.max_val
```

Comment:
Prefer following the API of the Normalization layer (mean and stddev normalization) to avoid issues when converting to a preprocessing layer.